tests/baselines/reference/ambientDeclarationsPatterns_merging1.errors.txt

```diff
@@ -0,0 +1,20 @@
+tests/cases/conformance/ambient/testB.ts(1,22): error TS2305: Module '"*.foo"' has no exported member 'onlyInA'.
+
+
+==== tests/cases/conformance/ambient/types.ts (0 errors) ====
+    declare module "*.foo" {
+      let everywhere: string;

```

Comment:
here's a terrifying test case

```
declare module "*.foo" {
  export interface OhNo { star: string }
}
declare module "a.foo" {
  export interface OhNo { a: string }
}
import { OhNo } from "b.foo"
declare let ohno: OhNo;
ohno.a // oh no
```

If we can give an error here, we should. Or we could try to recursively do what this PR does.

Reply:
😱 OH NO

Reply:
![image](https://user-images.githubusercontent.com/3277153/56674839-9b096d00-666f-11e9-8c5c-a18f04b722a8.png)

Doing the "unidirectional" thing recursively fixed this immediately. It feels like I'm on the right track, so ping me if you get a chance to elaborate on your concerns about symbol mutation, and hopefully I can address them in a way that continues to work well recursively like this.